app/main.py

```diff
@@ -1,213 +1,78 @@
-    # arthur
-    arthur = knightsConfig["arthur"]
-
-    # apply armour
-    arthur["protection"] = 0
-    for a in arthur["armour"]:
-        arthur["protection"] += a["protection"]
-
-    # apply weapon
-    arthur["power"] += arthur["weapon"]["power"]
-
-    # apply potion if exist
-    if arthur["potion"] is not None:
-        if "power" in arthur["potion"]["effect"]:
-            arthur["power"] += arthur["potion"]["effect"]["power"]
-
-        if "protection" in arthur["potion"]["effect"]:
-            arthur["protection"] += arthur["potion"]["effect"]["protection"]
-
-        if "hp" in arthur["potion"]["effect"]:
-            arthur["hp"] += arthur["potion"]["effect"]["hp"]
-
-    # mordred
-    mordred = knightsConfig["mordred"]
-
-    # apply armour
-    mordred["protection"] = 0
-    for a in mordred["armour"]:
-        mordred["protection"] += a["protection"]
-
-    # apply weapon
-    mordred["power"] += mordred["weapon"]["power"]
-
-    # apply potion if exist
-    if mordred["potion"] is not None:
-        if "power" in mordred["potion"]["effect"]:
-            mordred["power"] += mordred["potion"]["effect"]["power"]
-
-        if "protection" in mordred["potion"]["effect"]:
-            mordred["protection"] += mordred["potion"]["effect"]["protection"]
-
-        if "hp" in mordred["potion"]["effect"]:
-            mordred["hp"] += mordred["potion"]["effect"]["hp"]
-
-    # red_knight
-    red_knight = knightsConfig["red_knight"]
-
-    # apply armour
-    red_knight["protection"] = 0
-    for a in red_knight["armour"]:
-        red_knight["protection"] += a["protection"]
-
-    # apply weapon
-    red_knight["power"] += red_knight["weapon"]["power"]
-
-    # apply potion if exist
-    if red_knight["potion"] is not None:
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
-
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
-
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
-
-    # -------------------------------------------------------------------------------
-    # BATTLE:
+from app.config.config import KNIGHTS
+from app.knights.battle_preparation import Knight
+from app.knights.battle import fight
+
+
+def battle(knights_config: dict) -> dict:
+    lancelot = Knight(
+        name=knights_config.get("lancelot").get("name").lower(),
+        power=knights_config.get("lancelot").get("power"),
+        hp=knights_config.get("lancelot").get("hp"),
+        armour=knights_config.get("lancelot").get("armour"),
+        weapon=knights_config.get("lancelot").get("weapon"),
+        potion=knights_config.get("lancelot").get("potion"),
+        protection=sum(
+            [protect.get("protection")
+             for protect in knights_config.get("lancelot").get("armour")]
+        )
+    )
```

Comment:
```suggestion
    lancelot = Knight(**knights_config.get("lancelot"))
```

1. It's better to perform unpacking to reduce code duplication and improve its style.
2. Calculating protection is better moved to the Knight class, so it is done there and not repeated every time a new hero is created.

Reply:
fixed